File: src/main/java/burrows/apps/example/gif/di/module/SchedulerProviderModule.java
package burrows.apps.example.gif.di.module;


import burrows.apps.example.gif.di.scope.PerActivity;
import burrows.apps.example.gif.presentation.SchedulerProvider;
import dagger.Module;
import dagger.Provides;

/**
 * @author <a href="mailto:jaredsburrows@gmail.com">Jared Burrows</a>
 */
@Module
public final class SchedulerProviderModule {
  @Provides @PerActivity SchedulerProvider providerSchedulerProvider() {
    return new SchedulerProvider();
  }
}

File: src/main/kotlin/burrows/apps/example/gif/presentation/di/module/SchedulerProviderModule.kt
package burrows.apps.example.gif.presentation.di.module

import burrows.apps.example.gif.presentation.SchedulerProvider
import burrows.apps.example.gif.presentation.di.scope.PerActivity
import dagger.Module
import dagger.Provides

/**
 * @author [Jared Burrows](mailto:jaredsburrows@gmail.com)
 */
@Module
class SchedulerProviderModule {
  @Provides @PerActivity internal fun providerSchedulerProvider(): SchedulerProvider {
    return SchedulerProvider()
  }
}
